python/llm/src/bigdl/llm/transformers/models/gptj.py

```diff
@@ -100,28 +103,22 @@ def gptj_attention_forward(
     key = self._split_heads(key, self.num_attention_heads, self.head_dim, True)
     value = self._split_heads(value, self.num_attention_heads, self.head_dim, False)
 
-    if is_torch_fx_proxy(position_ids) or torch.jit.is_tracing():
-        # The logic to conditionally copy to GPU could not be traced, so we do this
-        # every time in the torch.fx case
-        embed_positions = get_embed_positions(self.embed_positions, position_ids)
-    else:
-        embed_positions = self._get_embed_positions(position_ids)
+    # if is_torch_fx_proxy(position_ids) or torch.jit.is_tracing():
+    #     # The logic to conditionally copy to GPU could not be traced, so we do this
+    #     # every time in the torch.fx case
+    #     embed_positions = get_embed_positions(self.embed_positions, position_ids)
+    # else:
+    #     embed_positions = self._get_embed_positions(position_ids)
 
-    repeated_position_ids = position_ids.unsqueeze(-1).repeat(1, 1, embed_positions.shape[-1])
-    sincos = torch.gather(embed_positions, 1, repeated_position_ids)
-    sin, cos = torch.split(sincos, sincos.shape[-1] // 2, dim=-1)
+    # repeated_position_ids = position_ids.unsqueeze(-1).repeat(1, 1, embed_positions.shape[-1])
+    # sincos = torch.gather(embed_positions, 1, repeated_position_ids)
+    # sin, cos = torch.split(sincos, sincos.shape[-1] // 2, dim=-1)
+    sin, cos = rotary_emb
 
     if self.rotary_dim is not None:
         k_rot = key[:, :, :, : self.rotary_dim]
-        k_pass = key[:, :, :, self.rotary_dim:]
-
         q_rot = query[:, :, :, : self.rotary_dim]
-        q_pass = query[:, :, :, self.rotary_dim:]
-
         q_rot, k_rot = apply_rotary_pos_emb(q_rot, k_rot, cos, sin, position_ids, "gptj")
```

Comment:
Add a comment for this? `torch.ops.torch_ipex.apply_rotary_embedding_two_qk` is inplacing.

Reply:
updated.